float64 bx
float64 by
float64 bz
float64 wx
float64 wy
float64 wz
int32 id
